examples/prom/maxwell_local_rom_greedy.cpp

```diff
@@ -0,0 +1,725 @@
+    args.AddOption(&online, "-online", "--online", "-no-online", "--no-online",
+                   "Enable or disable the online phase.");
+    args.AddOption(&build_database, "-build_database", "--build_database",
+                   "-no-build_database", "--no-build_database",
+                   "Enable or disable the build_database phase of the greedy algorithm.");
+    args.AddOption(&use_database, "-use_database", "--use_database",
+                   "-no-use_database", "--no-use_database",
+                   "Enable or disable the use_database phase of the greedy algorithm.");
+    args.AddOption(&greedy_param_space_min, "-greedy-param-min",
+                   "--greedy-param-min", "The minimum value of the parameter point space.");
+    args.AddOption(&greedy_param_space_max, "-greedy-param-max",
+                   "--greedy-param-max", "The maximum value of the parameter point space.");
+    args.AddOption(&greedy_param_space_size, "-greedy-param-size",
+                   "--greedy-param-size",
+                   "The number of values to search in the parameter point space.");
+    args.AddOption(&greedy_relative_error_tol, "-greedyrelerrortol",
+                   "--greedyrelerrortol", "The greedy algorithm relative error tolerance.");
+    args.AddOption(&greedy_subset_size, "-greedysubsize", "--greedysubsize",
+                   "The greedy algorithm subset size.");
+    args.AddOption(&greedy_convergence_subset_size, "-greedyconvsize",
+                   "--greedyconvsize", "The greedy algorithm convergence subset size.");
+
+    args.Parse();
+    if (!args.Good())
+    {
+        if (myid == 0)
+        {
+            args.PrintUsage(cout);
+        }
+        return 1;
+    }
+    if (myid == 0)
+    {
+        args.PrintOptions(cout);
+    }
+
+    if (fom)
+    {
+        MFEM_VERIFY(fom && !offline
+                    && !online, "everything must be turned off if fom is used.");
+    }
+
+    CAROM::GreedySampler* greedy_sampler = NULL;
+    MFEM_VERIFY(!build_database
+                || !use_database,
+                "both build_database and use_database can not be used at the same time.");
+
+    // 3a. Set up the ROM database for the greedy algorithm to run.
+    if (build_database)
+    {
+        MFEM_VERIFY(!offline
+                    && !online,
+                    "offline and online must be turned off during the build_database phase.");
+        MFEM_VERIFY(!visit
+                    && !visualization,
+                    "visit and visualization must be turned off during the build_database phase.")
+        std::ifstream infile("maxwell_local_rom_greedy_algorithm_data");
+        if (infile.good())
+        {
+            if (myid == 0) cout << "The database has already been built."
+                                    << "Exiting." << endl;
+            return 0;
+        }
+        infile.close();
+        greedy_sampler = new CAROM::GreedyRandomSampler(greedy_param_space_min,
```

Comment:
This pointer is not deleted.

Reply:
Good catch, I delete it now after running the do loop